code/obj/item/device/nuclear_op_items.dm

```diff
@@ -0,0 +1,824 @@
+		qdel(src)
+
+/obj/machinery/computer/security/pod_timer
+	maptext_x = 0
+	maptext_y = 20
+	maptext_width = 64
+	var/total_pod_time
+	processing_tier = PROCESSING_QUARTER
+
+	proc/get_pod_timer()
+		var/timeleft = round((total_pod_time - TIME) / 10, 1)
+		timeleft = "[(timeleft / 60) % 60]:[add_zero(num2text(timeleft % 60), 2)]"
+		return timeleft
+
+	process()
+		if (total_pod_time && TIME >= total_pod_time)
+			src.maptext = "<span style=\"color: red; font-family: Fixedsys, monospace; text-align: center; vertical-align: top; -dm-text-outline: 1 black;\">--:--</span>"
+		else
+			src.maptext = "<span style=\"color: red; font-family: Fixedsys, monospace; text-align: center; vertical-align: top; -dm-text-outline: 1 black;\">[get_pod_timer()]</span>"
+		..()
+
+
+#define DESIGNATOR_MAX_RANGE 30
+
+////////// Laser Designator & Airstrikes //////////
+/obj/item/device/laser_designator
+	name = "Laser Designator"
+	icon = 'icons/obj/items/device.dmi'
+	desc = "A handheld monocular device with a laser built into it, used for calling in fire support."
+	icon_state = "laser_designator"
+	item_state = "electronic"
+	density = FALSE
+	anchored = FALSE
+	w_class = W_CLASS_SMALL
+	/// How many times can this be used?
+	var/uses = 1
+	/// Movement controller for the designator's "scope"
+	var/datum/movement_controller/designatormove = null
+	/// TRUE if an air strike is waiting to happen/happening
+	var/in_use = FALSE
+	/// The gun that "fires" the shell
+	var/obj/machinery/broadside_gun/linked_gun = null
+	/// Takes a string for a ship that's set in the `linked_gun`'s vars, e.g. "Cairngorm"
+	var/ship_looking_for = ""
+	/// Overlay sprite for where the strike will land, set to null for no overlay
+	var/image/target_overlay = null
+
+	New()
+		..()
+		designatormove = new/datum/movement_controller/designator_look()
+		desc = "A handheld monocular device with a laser built into it, used for calling in fire support. It has [src.uses] charge left."
+		target_overlay = image('icons/effects/effects.dmi', "spinny_red")
+
+	disposing()
+		designatormove = null
+		linked_gun = null
+		target_overlay = null
+		..()
+
+	dropped(mob/M)
+		remove_self(M)
+		..()
+
+	move_callback(var/mob/living/M, var/turf/source, var/turf/target)
+		if (M.use_movement_controller)
+			if (source != target)
+				just_stop_designating(M)
+
+	proc/airstrike(atom/target, params, mob/user, reach)
+		uses -= 1
+		in_use = TRUE
+		if(!linked_gun.bombard(target, user))
+			uses += 1
+		in_use = FALSE
+
+	proc/remove_self(var/mob/living/M)
+		if (islist(M.move_laying))
+			M.move_laying -= src
+		else
+			M.move_laying = null
+
+		if (ishuman(M))
+			M:special_sprint &= ~SPRINT_DESIGNATOR
+
+		just_stop_designating(M)
+
+	proc/just_stop_designating(var/mob/living/M) // remove overlay here
+		if (M.client)
+			M.client.pixel_x = 0
+			M.client.pixel_y = 0
+
+		M.use_movement_controller = null
+		M.keys_changed(0,0xFFFF)
```

Comment:
need a comment on this

Reply:
I don't know _exactly_ how this works, it was just code inherited from the sniper's code, but it breaks without it (at least in my testing)